compiler/qsc/src/bin/qsc.rs

```diff
@@ -95,64 +96,91 @@ enum Emit {
     Qir,
 }
 
+#[allow(clippy::too_many_lines)]
 fn main() -> miette::Result<ExitCode> {
     env_logger::init();
     let cli = Cli::parse();
-    let mut store = PackageStore::new(qsc::compile::core());
-    let mut dependencies = Vec::new();
     let profile: qsc::target::Profile = cli.profile.unwrap_or_default().into();
     let capabilities = profile.into();
     let package_type = if cli.emit.contains(&Emit::Qir) {
         PackageType::Exe
     } else {
         PackageType::Lib
     };
-
-    if !cli.nostdlib {
-        dependencies.push(store.insert(qsc::compile::std(&store, capabilities)));
-    }
-
     let mut features = LanguageFeatures::from_iter(cli.features);
 
-    let mut sources = cli
-        .sources
-        .iter()
-        .map(read_source)
-        .collect::<miette::Result<Vec<_>>>()?;
-
-    if sources.is_empty() {
-        let fs = StdFs;
-        if let Some(qsharp_json) = cli.qsharp_json {
-            if let Some(dir) = qsharp_json.parent() {
-                let project = match fs.load_project(dir, None) {
-                    Ok(project) => project,
-                    Err(errs) => {
-                        for e in errs {
-                            eprintln!("{e:?}");
-                        }
-                        return Ok(ExitCode::FAILURE);
+    let (mut store, dependencies, source_map) = if let Some(qsharp_json) = cli.qsharp_json {
```

Comment:
This block of logic grew considerably in this PR. It would be worth breaking it down in different functions so it can be more easily seen where `store`, `dependencies` and `source_map` come from.